World.py

```diff
@@ -0,0 +1,156 @@
+import random
+from Game import Strategy
+from Board import RectangularGrid
+
+# delete later
+import time
+
+class World():
+    def __init__(self, game, board, players, r = 0, q = 0, noise1 = False, noise2 = False, imitation = False, migration = False, moore_neighborhood_range = 0):
+        """
+        - game: the game played between two players during an interaction
+        - board: the topology of the world
+        - players: a list of players in the world
+        - r: the probability of a player randomly resetting its strategy
+        - q: the probability that a player resets its strategy to cooperate
+        - noise1: a boolean indicating whether Noise 1 is present
+        - noise2: a boolean indicating whether Noise 2 is present
+        - imitation: a boolean indicating whether players perform imitation
+        - migration: a boolean indicating whether
+        - moore_neighborhood_range: the range of the Moore neighborhood around each cell
+        """
+        self.game = game
+        self.board = board
+        self.players = players
+        self.r = r
+        self.q = q
+        self.noise1 = noise1
+        self.noise2 = noise2
+        self.imitation = imitation
+        self.migration = migration
+        self.moore_neighborhood_range = moore_neighborhood_range
+
+        # randomly insert players in board cells
+        random.shuffle(players)
+        random_cell_sequence = board.random_cell_sequence()
+        for i in range(len(players)):
+            players[i].cell = random_cell_sequence[i]
+            self.board.set_cell(random_cell_sequence[i], players[i])
+            self.board.get_cell(random_cell_sequence[i])
```

Comment:
note: if we have more players than cells we will get an eeor

Reply:
Good point. I'll add in a length check before the players are assigned to the grid.